File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int n = nums.size();
        int i = 0;
        int j = n - 1;
        int m;
        while (i < j) {
            m = i + (j - i) / 2;
            if (nums[m] == target) {
                return m;
            }
            else if (nums[m] < target) {
                i = m + 1;
            }
            else {
                j = m - 1;
            }
        }
        if (j < 0) {
            return 0;
        }
        if (nums[i] < target) {
            return i + 1;
        }
        return i;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        n = len(nums)
        i = 0
        j = n - 1
        while i < j:
            m = (i + j) // 2
            if nums[m] == target:
                return m
            elif nums[m] < target:
                i = m + 1
            else:
                j = m - 1
        if j < 0:
            return 0
        if nums[i] < target:
            return i + 1
        return i
